src/github/pullRequestManager.ts

```diff
@@ -119,6 +124,47 @@ export class PullRequestManager implements IPullRequestManager {
 		}
 	}
 
+	private validateRemotesSetting() {
+		const remotesSetting = vscode.workspace.getConfiguration(SETTINGS_NAMESPACE).get<string[]>(REMOTES_SETTING);
+
+		if (remotesSetting) {
+			return remotesSetting.forEach(remote => {
+				if (!this._githubRepositories.some(repo => repo.remote.remoteName === remote)) {

```

Comment:
Since this setting can be non-workspace setting, if I put `upstream` and `origin` in my user setting, it will show warning to most of my repositories. Probably we should fail slightly (but emitting log to output).